ledger/sandbox-on-x/src/main/scala/com/daml/ledger/sandbox/bridge/validate/SequencerState.scala

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.sandbox.bridge.validate
+
+import com.daml.ledger.offset.Offset
+import com.daml.ledger.sandbox.bridge.BridgeMetrics
+import com.daml.ledger.sandbox.bridge.validate.SequencerState.{LastUpdatedAt, SequencerQueue}
+import com.daml.lf.transaction.GlobalKey
+import com.daml.platform.store.appendonlydao.events.{ContractId, Key}
+
+import scala.collection.Searching
+import scala.util.chaining._
+
+/** State of the sequencing conflict checking stage
+  *
+  * @param sequencerQueue A queue ordered by each transaction's offset.
```

Comment:
I would consider creating a smart constructor to ensure that a vector passed as `sequencerQueue` is ordered. Otherwise the ordering invariant can be violated.

Reply:
Alright. Did so: the `SequencerState` can only be constructed empty